crypto/bn/bn_print.c

```diff
@@ -32,14 +32,14 @@ char *BN_bn2hex(const BIGNUM *a)
     }
     p = buf;
     if (a->neg)
-        *(p++) = '-';
+        *p++ = '-';
     for (i = a->top - 1; i >= 0; i--) {
         for (j = BN_BITS2 - 8; j >= 0; j -= 8) {
             /* strip leading zeros */
-            v = ((int)(a->d[i] >> (long)j)) & 0xff;
-            if (z || (v != 0)) {
-                *(p++) = Hex[v >> 4];
-                *(p++) = Hex[v & 0x0f];
+            v = (int)(0xff & (a->d[i] >> j));
```

Comment:
Why swap the `&` operands?  Considering the rest of OpenSSL uses the `var & 0xwhatever` form, this one looks odd...